sklearn/metrics/pairwise.py

```diff
@@ -585,6 +592,119 @@ def cosine_distances(X, Y=None):
     return S
 
 
+def gower_distances(X, Y=None, w=None, categorical_features=None):
+    """
+    Computes the gower distances between X and Y
+
+    Read more in the :ref:`User Guide <metrics>`.
+
+    Parameters
+    ----------
+    X : array-like, shape (n_samples, n_features)
+
+    Y : array-like, shape (n_samples, n_features)
+
+    w:  array-like, shape (n_features)
+    According the Gower formula, w is an attribute weight.
+
+    categorical_features: array-like, shape (n_features)
+    Indicates with True/False wheter a column is a categorical attribute.
+    This is useful when categorical atributes are represented as integer
+    values.
+
+    Returns
+    -------
+    similarities : ndarray, shape (n_samples, )
+
+    Notes
+    ------
+    Gower is a similarity measure for categorical, boolean and numerical mixed
+    data.
+
+    """
+
+    X, Y = check_pairwise_arrays(X, Y, dtype=(np.object, None)[issparse(X) or
+                                                               issparse(Y)])
+
+    rows, cols = X.shape
+
+    if categorical_features is None:
+        categorical_features = []
+        for col in range(cols):
+            if np.issubdtype(type(X[0, col]), np.number):
+                categorical_features.append(False)
+            else:
+                categorical_features.append(True)
+    # Calculates the normalized ranges and max values of numeric values
+    ranges_of_numeric = [0.0] * cols
+    max_of_numeric = [0.0] * cols
+    for col in range(cols):
+        if not categorical_features[col]:
+            max = None
+            min = None
+            if issparse(X):
+                col_array = X.getcol(col)
+                max = col_array.max() + 0.0
+                min = col_array.min() + 0.0
+            else:
+                col_array = X[:, col].astype(np.double)
+                max = np.nanmax(col_array)
+                min = np.nanmin(col_array)
+
+            if np.isnan(max):
+                max = 0.0
+            if np.isnan(min):
+                min = 0.0
+            max_of_numeric[col] = max
+            ranges_of_numeric[col] = (1 - min / max) if (max != 0) else 0.0
```

Comment:
it seems strange to call a ratio a range. I've not reviewed the Gower formula yet, but this looks a bit strange, particularly in how it's affected by the sign of min and max.